src/lib/containers/EntityLink.tsx

```diff
@@ -1,31 +1,66 @@
+import { Skeleton } from '@material-ui/lab'
 import React from 'react'
-import { getEntityTypeFromHeader } from '../utils/functions/EntityTypeUtils'
+import {
+  convertToEntityType,
+  getEntityTypeFromHeader,
+} from '../utils/functions/EntityTypeUtils'
 import { PRODUCTION_ENDPOINT_CONFIG } from '../utils/functions/getEndpoint'
-import { EntityHeader } from '../utils/synapseTypes/'
+import { useGetEntity } from '../utils/hooks/SynapseAPI/useEntity'
+import { Entity, EntityHeader } from '../utils/synapseTypes/'
 import { EntityTypeIcon } from './EntityIcon'
 
 type EntityLinkProps = {
-  entityHeader: EntityHeader
+  entity: string | EntityHeader | Entity
+  link?: boolean
   className?: string
 }
 
-export const EntityLink: React.FC<EntityLinkProps> = ({
-  entityHeader,
-  className,
-}) => {
-  const { id, name } = entityHeader
-  const type = getEntityTypeFromHeader(entityHeader)
-  return (
-    <p className={className}>
-      <a
-        className={className}
-        target="_blank"
-        rel="noopener noreferrer"
-        href={`${PRODUCTION_ENDPOINT_CONFIG.PORTAL}#!Synapse:${id}`}
-      >
-        <EntityTypeIcon type={type} style={{ marginRight: '5px' }} />
-        {name}
-      </a>
-    </p>
-  )
+export const EntityLink = (props: EntityLinkProps) => {
```

Comment:
Will rename this later...didn't want to get logic changes lost in the diff due to a file rename